### PR TITLE
Improve `SHOW FUNCTIONS`

### LINKED ISSUE #3913: Show UDF definition by `SHOW FUNCTIONS`
Currently we have 2 commands to show UDF and show builtin functions, consider to combine them, which will produce this information:
```
name is_builtin definition description
```

### PR DESCRIPTION
**Pull request by lianghanzhen**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Merge `show udf` with `SHOW FUNCTIONS` which we can search both builtin & user defined functions.

- Show UDF for `SHOW FUNCTIONS` query;
- Remove `show udf` interpreter;
- Add docs for function features;

## Changelog

- Improvement

## Related Issues

Fixes #3913

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
Cool, thank you @lianghanzhen !
The unit-test is borken caused by github ci oom, it's not related to this PR, I will check it locally today.

**Comment by BohuTANG** [comment]
```
---- storages::system::functions_table::test_functions_table stdout ----
thread 'storages::system::functions_table::test_functions_table' panicked at 'assertion failed: `(left == right)`
  left: `5`,
 right: `2`', query/tests/it/storages/system/functions_table.rs:33:5

```

File: tests/suites/2_management_mode/00_0001_management_mode.sql [hunk 1]
```diff
@@ -30,14 +30,10 @@ DROP STAGE test_stage;
 -- udf check.
 CREATE FUNCTION xy AS (x, y) -> (x + y) / 2;
 ALTER FUNCTION xy AS (x, y) -> (x + y) / 3;
-SHOW FUNCTION xy;
```
Comment (BohuTANG) [inline tests/suites/2_management_mode/00_0001_management_mode.sql:33]: show functions like 'xy';

File: tests/suites/2_management_mode/00_0001_management_mode.sql [hunk 2]
```diff
@@ -30,14 +30,10 @@ DROP STAGE test_stage;
...
 DROP FUNCTION xy;
 
 
 -- TENANT2
 SUDO USE TENANT 'tenant2';
 SHOW DATABASES;
 SHOW TABLES;
-
--- check
-SHOW FUNCTION xy; -- {ErrorCode 2602}
```
Comment (BohuTANG) [inline tests/suites/2_management_mode/00_0001_management_mode.sql:43]: show functions like 'xy';

This is probe check, we should keep it :)

**Comment by BohuTANG** [comment]
@mergify update

**Comment by lianghanzhen** [comment]
Fix according to your comments @BohuTANG

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
@lianghanzhen Thank you!

**Pull request merged**